src/back-end/lib/crud.ts

```diff
@@ -118,6 +126,42 @@ export type Delete<
   "validateRequestBody" | "respond"
 >;
 
+export type BasicCrudResource<Session, Connection> = Resource<
+  DefaultSupportedRequestBodies,
+  DefaultSupportedResponseBodies,
+  any,
+  any,
+  any,
+  any,
+  any,
+  any,
+  any,
+  any,
+  any,
+  any,
+  Session,
+  Connection
+>;
+
+/**
+ * The Resource type for the back-end framework. Defines supported request/response bodies, CRUD operations,
+ * and database connection
+ *
+ * @typeParam SupportedRequestBodies - supported request bodies for this resource
+ * @typeParam SupportedResponseBodies - supported response bodies for this resource
+ * @typeParam CreatedParsedReqB - parsed request body for the Create operation
+ * @typeParam CreateValidatedReqB - validated request body for the Create operation
+ * @typeParam CreateReqBErrors - validation errors for the Create operation request body
+ * @typeParam ReadOneValidatedReqB - validated request body for the Read operation
+ * @typeParam ReadOneReqBErrors - validation errors for the Read operation request body
+ * @typeParam UpdateParsedReqB - parsed request body for the Update operation
+ * @typeParam UpdateValidatedReqB - validated request body for the Update operation
+ * @typeParam UpdateReqBErrors - validation errors for the Update operation request body
+ * @typeParam DeleteValidatedReqB - validated request body for the Delete operation
+ * @typeParam DeleteReqBErrors - validation errors for the Delete operation request body
+ * @typeParam Session - session type for this resource
+ * @typeParam Connection - database connection type for this resource
+ */
```

Comment:
annotating types this way is helpful. Would recommend applying this generally to more types, functions, and interfaces which adds to understand-ability.

Reply:
I can add additional TypeDocs to this PR, however I think it may be beneficial to have a separate PR with TypeDoc comments so that it can be reviewed independently. This PR is already getting a bit large. Happy to proceed with whichever way you feel is best @bdolor, just let me know.

Reply:
I hear ya. Would like to work towards a future state that includes doc comments as part of a PR, knowing how difficult prioritizing documentation as a separate chunk of work can be. In the spirit of moving this PR along, please indulge the following:

the output of `npm run typedoc:build` and `npm run typedoc:start` automatically recognizes and produces light (nearly useless) docs for three methods in the backend/index.js: `connectToDatabase`, `createDowntimeRouter` and `createRouter`. Example: 

![image](https://user-images.githubusercontent.com/2048170/195712761-39b9f665-8143-4427-9622-2d899b8c8b8b.png)

when the following doc comment is added

![image](https://user-images.githubusercontent.com/2048170/195712979-d10c5f63-0ed3-44e5-9e30-d3b921c9c4f1.png)

it creates redundant information to the auto-generated documentation. 
 
![image](https://user-images.githubusercontent.com/2048170/195706358-59460218-6c65-493b-bb8e-6ff532ab062d.png)
 
which is to say, at some point I'd like to figure out how to bridge the gap between expectations and what's being generated.